### PR TITLE
Slight general cleanup, enable dma-macros test, allow using virtual mem2mem channel on c2

### PR DESCRIPTION
**Pull request by bugadani**
- Record MSRV change in embassy crate
- change cfgs to peripherals
- use const blocks instead of interior mutable consts
- remove some unnecessary internal type aliasses
- enables `dma_macros` for ESP32 and S2
- allow using MEM2MEM1 on ESP32C2. The DMA seems to ignore the peripheral selection bit for in-memory transactions

### CODE REVIEW

File: hil-test/tests/dma_mem2mem.rs [hunk 1]
```diff
@@ -17,10 +17,18 @@ use hil_test as _;
...
-    if #[cfg(any(feature = "esp32c2", feature = "esp32c3", feature = "esp32s3"))] {
+    if #[cfg(any(esp32c2, esp32c6, esp32h2))] {
+        type DmaPeripheralType = esp_hal::peripherals::MEM2MEM1;
+    } else {
         type DmaPeripheralType = esp_hal::peripherals::SPI2;
+    }
+}
+
+cfg_if::cfg_if! {
+    if #[cfg(any(esp32, esp32s2))] {
```
Comment (bjoernQ) [inline hil-test/tests/dma_mem2mem.rs:28]: ESP32 / S2 are not supported - no need for this

Comment (bugadani) [reply hil-test/tests/dma_mem2mem.rs:28]: not yet!

File: esp-hal/src/i2c.rs [hunk 2]
```diff
@@ -533,23 +533,14 @@ mod asynch {
...
-        if #[cfg(all(i2c0, i2c1))] {
-            const NUM_I2C: usize = 2;
-        } else if #[cfg(i2c0)] {
-            const NUM_I2C: usize = 1;
-        }
-    }
-    #[allow(clippy::declare_interior_mutable_const)]
-    const INIT: AtomicWaker = AtomicWaker::new();
-    static WAKERS: [AtomicWaker; NUM_I2C] = [INIT; NUM_I2C];
+    const NUM_I2C: usize = 1 + cfg!(i2c1) as usize;
```
Comment (MabezDev) [inline esp-hal/src/i2c.rs:542]: This is a nice trick, we should do this in more places.

**Review by MabezDev — APPROVED** [review]
If this is just meant to be a clean up PR, LGTM!

File: esp-hal-embassy/src/time_driver.rs [hunk 3]
```diff
@@ -36,11 +36,8 @@ pub(super) struct EmbassyTimer {
     alarms: Mutex<[AlarmState; MAX_SUPPORTED_ALARM_COUNT]>,
 }
 
-#[allow(clippy::declare_interior_mutable_const)]
-const ALARM_STATE_NONE: AlarmState = AlarmState::new();
-
 embassy_time_driver::time_driver_impl!(static DRIVER: EmbassyTimer = EmbassyTimer {
-    alarms: Mutex::new([ALARM_STATE_NONE; MAX_SUPPORTED_ALARM_COUNT]),
+    alarms: Mutex::new([const { AlarmState::new() }; MAX_SUPPORTED_ALARM_COUNT]),
```
[the remark this answers is not in the document]
Comment (bugadani) [reply esp-hal-embassy/src/time_driver.rs:40]: New toys with the 1.79 msrv :)

File: esp-hal/src/dma/pdma.rs [hunk 4]
```diff
@@ -57,20 +47,9 @@ macro_rules! ImplSpiChannel {
...
-                                .modify(|_, w| unsafe { w.spi2_dma_chan_sel().bits(1) });
-                            },
-                            3 => {
-                                dport
-                                .spi_dma_chan_sel()
-                                .modify(|_, w| unsafe { w.spi3_dma_chan_sel().bits(2) });
-                            },
-                            _ => panic!("Only SPI2 and SPI3 supported"),
-                        }
+                        dport
```
Comment (Dominaezzz) [inline esp-hal/src/dma/pdma.rs:50]: This `init_channel` function can be removed and this code can be moved to either `configure`/`configure_for_async` or `set_peripheral`.

Comment (bugadani) [reply esp-hal/src/dma/pdma.rs:50]: I didn't want to modify this code too much lest your PR will probably conflict, but let's see :)

**Review by jessebraham — APPROVED** [review]

**Comment by bugadani** [comment]
Failure looks like a network error if anything; rebased & retrying merge

**Comment by jessebraham** [comment]
> Failure looks like a network error if anything; rebased & retrying merge

Yeah I've seen this happen a few times before, has always succeeded in a subsequent run in my experience at least.

**Pull request merged**